SharkCage/CageManager/SecuritySetup.h

```diff
@@ -5,7 +5,10 @@
 #include <string>
 
 template<typename T>
-auto local_free_deleter = [&](T resource) { ::LocalFree(resource); };
+auto local_free_deleter = [&](T *resource) { ::LocalFree(resource); };
+
+typedef void Sid;
+typedef std::unique_ptr<Sid, decltype(local_free_deleter<Sid>)> SidPointer;
```

Comment:
Should this be so prcisely instantiated? `::LocalFree` is not the only function used to deallocate `PSID`. On line `114` you use `::AllocateAndInitializeSid` function to build `SID` for `BUILTIN\Administrators`. This SID should be deallocated by `::FreeSid` function. (Currently there is a resource leak - I fied it in #105 by calling `::FreeSid` on each exit route). This should IMO be handled by smart pointer as well. Maybe deleter could stay as a template parameter?